virt-v2v/cold/entrypoint.go

```diff
@@ -169,16 +175,26 @@ func LinkDisks(diskKind string, num int) (err error) {
 	return
 }
 
-func executeVirtV2v(args []string) (err error) {
+func executeVirtV2v(args []string, source string) (err error) {
 	virtV2vCmd := exec.Command(args[0], args[1:]...)
 	virtV2vStdoutPipe, err := virtV2vCmd.StdoutPipe()
 	if err != nil {
 		fmt.Printf("Error setting up stdout pipe: %v\n", err)
 		return
 	}
+	teeOut := io.TeeReader(virtV2vStdoutPipe, os.Stdout)
 
-	tee := io.TeeReader(virtV2vStdoutPipe, os.Stdout)
-	virtV2vCmd.Stderr = os.Stderr
+	var teeErr io.Reader
+	if source == OVA {
+		virtV2vStderrPipe, err := virtV2vCmd.StderrPipe()
+		if err != nil {
+			fmt.Printf("Error setting up stdout pipe: %v\n", err)
+			return err
+		}
+		teeErr = io.TeeReader(virtV2vStderrPipe, os.Stderr)
```

Comment:
what's the benefit of it? will the stderr will still be shown in the logs?

i see you use it later on for scanning - do we expect the see the expected value in an error?

Reply:
Yes, all the extra detailed log of virt-v2v is directed to stdErr so we will see the firmware detection there, the virt-v2v standard output is only the stages of the conversion and the disk copy percentages.

Reply:
here is an example of the command output to understand it better, all other information is redirected to `Stderr`

`[   0.0] Setting up the source: -i ova /home/bkhizgiy/local_code_test/rhel9-efi.ova
virt-v2v: warning: manifest has a checksum for non-existent file 1nisim-rhel9-efi-2.iso (ignored)
[   6.5] Opening the source
[  10.6] Inspecting the source
[  14.1] Checking for sufficient free disk space in the guest
[  14.1] Converting Red Hat Enterprise Linux 9.3 Beta (Plow) to run on KVM
virt-v2v: The QEMU Guest Agent will be installed for this guest at first boot.
virt-v2v: This guest has virtio drivers installed.
[  67.7] Mapping filesystem data to avoid copying unused and blank areas
[  68.3] Closing the overlay
[  68.5] Assigning disks to buses
[  68.5] Checking if the guest needs BIOS or UEFI to boot
[  68.5] Setting up the destination: -o disk -os /var/tmp/v2v
[  69.6] Copying disk 1/1
[  86.6] Creating output metadata
[  86.6] Finishing off
`

Reply:
this is an example output after the change?
we do wish to have the whole logs

Reply:
nono, only for the Stdout, so it would be more clear why we need to scan the Stderr (since all the information is there :)), the pod output will remain the same as it was.